aplib.net-demo/Assets/Scripts/Points/EnemyPointsAdder.cs

```diff
@@ -0,0 +1,22 @@
+using UnityEngine;
+
+[RequireComponent(typeof(HealthComponent))]
+public class EnemyPointsAdder : PointsAdderComponent
+{
+    private HealthComponent _enemyHealth;
+
+    /// <summary>
+    /// Grab the enemy health component.
+    /// </summary>
+    private void Awake()
+    {
+        _enemyHealth = GetComponent<HealthComponent>();
+        _enemyHealth.Death += SendPoints;
+    }
```

Comment:
You should be subscribing to and unsubscribing from events in OnEnable() and OnDisable.